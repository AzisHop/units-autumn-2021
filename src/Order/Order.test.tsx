import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {OrderComponent} from './Order';
import {fakeOrders} from '../data/fakeOrders';

jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter()});

describe('Order.tsx', () => {
	beforeEach(() => {
		(getDate as jest.Mock).mockReturnValue('300');
	});

	afterEach(() => {
		jest.resetAllMocks();
	});

	it('right render', () => {
		const wrapper = shallow(<OrderComponent order={fakeOrders[0]}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('render without items', () => {
		const wrapper = shallow(<OrderComponent order={
			{
				id: 123,
				date: 300,
				shop: 'Bizarro',
			}
		}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('render without shop', () => {
		const wrapper = shallow(<OrderComponent order={
			{
				id: 1,
				date: 300,
				items: [
					'item1',
					'item2'
				]
			}
		}/>);
		expect(wrapper.isEmptyRender()).toBeTruthy();
	});

	it('render with shop and items', () => {
		const wrapper = shallow(<OrderComponent order={
			{
				id: 1,
				date: 300,
				shop: 'Bizarro',
				items: [
					'item1',
					'item2'
				]
			}
		}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('render without order', () => {
		const wrapper = shallow(<OrderComponent order={{}}/>);
		expect(wrapper.isEmptyRender()).toBeTruthy();
	});

	it('check if getDate is called', () => {
		const testData = {shop: 'shop', date: 1};
		shallow(<OrderComponent order={testData}/>);

		expect(getDate).toHaveBeenCalledWith(testData.date);
	});
});
