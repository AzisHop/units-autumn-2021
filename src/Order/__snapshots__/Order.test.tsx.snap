// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.tsx render with shop and items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Bizarro
    </span>
    <span
      id="date"
    >
      300
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      item1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      item2
    </div>
  </div>
</div>
`;

exports[`Order.tsx render without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Bizarro
    </span>
    <span
      id="date"
    >
      300
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order.tsx right render 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      300
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;
